### PR TITLE
Global menu

### PR DESCRIPTION
**Pull request by svartkanin**
This is a follow-up PR of the previously merged new interface menu (https://github.com/archlinux/archinstall/pull/660).

Bare in mind that this is still WIP (there are some minor things left to fix) but I wanted to raise it to get some feedback on it. 

@Torxed as we discussed in the previous PR we wanted to introduce an overall menu as well where previously selected options could be changed again. 
I've added a global menu now with pre-set defaults as they're used in the single menus, so that the user doesn't have to modify anything they're happy with in the first place. 

In addition, the defaults get overridden by any value that has been specified via a config (basically anything that has been stored in the `archinstall.arguments` container. This way those values are already set and displayed and can, if needed, also be changed again. 

I'll update the PR to fix the remaining things but it is mostly functioning and ready to play around with

### CODE REVIEW

**Comment by svartkanin** [comment]
@Torxed I've updated the last bits and pieces and it should be ready for a review now

File: examples/guided.py
Comment (Torxed) [inline examples/guided.py:75]: You can omit the `from archinstall.lib.menu.selection_menu import GlobalMenu` and use `archinstall.GlobalMenu` as long as you expose the class in: https://github.com/archlinux/archinstall/blob/577a060e792a82e1e93822c9a16b4093f94c003c/archinstall/__init__.py#L23
This will make the guided example a bit more cleaner. As it serves as an example of how to write an installer not having to dig through the inners of archinstall and learn all the paths to stuff :)

**Review by Torxed — CHANGES_REQUESTED** [review]
I only have one minor tweak, then I'm ready to merge :) Solid work on this one, guided became a lot cleaner :)

File: examples/guided.py
Comment (svartkanin) [reply examples/guided.py]: @Torxed awesome thanks for the review! 
I've updated the import

Comment (Torxed) [reply examples/guided.py]: Great work! We've modified guided a bit in regards to `"!users"` and `"harddrives"` section, could you pull that in to the global menu? I think that would help with the merge conflict against master that is right now.

**Comment by svartkanin** [comment]
I realized the conflicts only after, I think I had the user changes already in there, working on the harddrive now

**Comment by svartkanin** [comment]
@Torxed I fixed the conflicts now (hopefully correctly ;)) and updated minor dependency things in the menu options

**Comment by Torxed** [comment]
I'll run a last test probably tomorrow and if everything works (which I'm sure it will) this will fly into `master` and we have ourselves a new menu system :) Great work so far!

**Comment by wllacer** [comment]
@svartkanin Good work. It looks very beautiful now.
After playing a while with it, i have some suggestions. Where do you want that i send them, here or at your repository ?. But two merit to be here
**A bug report:**. If you are working with merge have a look at some changes at guided.load_config where the disk_layouts file is loaded. Since a couple of days, this is not needed anymore (it's loaded at __init__.py and with a seemly less error prone code). There, testing your branch, did reproduce a well know bug, we were already rid.
**A wish**, please don't delete -still- the old user interfaces. Some workflows might not be ready to be changed on short notice.

**Comment by svartkanin** [comment]
@wllacer feel free to leave any feedback here in this thread, I'll try to address it when time allows

**Comment by Torxed** [comment]
I'm ready to merge this. I'll probably hold off on #846 until this is merged as I would probably poke around in too many places where this interacts.

**Comment by svartkanin** [comment]
@Torxed sounds good!

**Comment by Torxed** [comment]
Great work! Any remaining issues can be done in a new PR :) We probably need to tweak a bit :)

**Comment by svartkanin** [comment]
@Torxed Fantastic :partying_face: , please feel free to assign any tweaks to me if you like, I'm happy to take them on since I got the context :)

**Comment by Torxed** [comment]
Will do with pleasure! hehe

**Pull request merged**
